src/main/scala/lc/Main.scala

```diff
@@ -14,37 +24,54 @@ trait ChallengeProvider {
 class Captcha {
   val con: Connection = DriverManager.getConnection("jdbc:h2:./captcha", "sa", "")
   val stmt: Statement = con.createStatement()
-  stmt.execute("CREATE TABLE IF NOT EXISTS challenge(token varchar, id varchar, secret varchar, image blob)")
-  val insertPstmt: PreparedStatement = con.prepareStatement("INSERT INTO challenge(token, id, secret) VALUES (?, ?, ?)")
-  val selectPstmt: PreparedStatement = con.prepareStatement("SELECT secret FROM challenge WHERE token = ?")
-  
-  def getCaptcha(): Boolean = {
-    val provider = new FilterChallenge
-    val (token, image) = this.getChallenge(provider)
-    image.output(new File("Captcha.png"))
-    println(s"Token: ${token}")
-    println("Enter your answer: ")
-    val answer = scala.io.StdIn.readLine()
-    this.getAnswer(token, answer, provider)
+  stmt.execute("CREATE TABLE IF NOT EXISTS challenge(token varchar, id varchar, secret varchar, provider other, image blob)")
+  val insertPstmt: PreparedStatement = con.prepareStatement("INSERT INTO challenge(token, id, secret, provider, image) VALUES (?, ?, ?, ?, ?)")
+  val selectPstmt: PreparedStatement = con.prepareStatement("SELECT secret, provider FROM challenge WHERE token = ?")
+  val imagePstmt: PreparedStatement = con.prepareStatement("SELECT image FROM challenge WHERE token = ?")
+
+  def getCaptcha(id: Id): Array[Byte] = {
+  	imagePstmt.setString(1, id.id)
+  	val rs: ResultSet = imagePstmt.executeQuery()
+  	rs.next()
+  	val blob = rs.getBlob("image")
+  	var image :Array[Byte] = null
+  	if(blob != null)
+  		image =  blob.getBytes(1, blob.length().toInt)
+  	image
+  }
+
+  def convertImage(image: Image): ByteArrayInputStream = {
+  	val output = new ByteArrayOutputStream()
+  	image.output(new File("Captcha.png"))
+  	val img = ImageIO.read(new File("Captcha.png"))
+  	ImageIO.write(img,"png",output)
+  	val byte_array = output.toByteArray()
+  	val blob = new ByteArrayInputStream(byte_array)
+  	blob
   }
   
-  def getChallenge(provider: ChallengeProvider): (String, Image) = {
+  def getChallenge(param: Parameters): Id = {
+  	val provider = new FilterChallenge
```

Comment:
The `getChallenge` method was being called from `getCaptcha` method, which passed the provider to getChallenge. 
I have updated both the methods as per the end points. I am passing the `params` to `getChallenge` so that an appropriate provider is selected based on the captcha parameters.

Reply:
> an appropriate provider is selected based on the captcha parameters

Is that a TODO for later? Because, I can't find the selection logic.